### PR TITLE
make custom fields configurable

### PR DESCRIPTION
**Pull request by dleusing**

### CODE REVIEW

**Comment by dleusing** [comment]
CLA signed.

**Comment by bobapple** [comment]
@cla-bot check

**Review by dleusing — COMMENTED** [review]
fix for Whitespace found at end of line

File: application/clicommands/SendCommand.php [hunk 1]
```diff
@@ -114,11 +118,12 @@ public function problemAction()
             $ackMessage = "JIRA issue $key has been created";
         } else {
             $key = $issue->key;
-            $currentStatus = isset($issue->fields->icingaStatus) ? $issue->fields->icingaStatus : null;
+            $icingaStatus = $config->get('ui', 'field_icingaStatus', 'customfield_19220');
```
Comment (Thomas-Gelf) [inline application/clicommands/SendCommand.php:121]: Guess something went wrong here. It's not UI, and we cannot hard-code custom field identifiers - they are different on different Jira instances.

File: library/Jira/RestApi.php [hunk 2]
```diff
@@ -125,22 +127,24 @@ public function updateIssue(IssueUpdate $update)
 
     protected function prepareIssueQuery($host = null, $service = null, $onlyOpen = true)
     {
+        $config = Config::module('jira');
```
Comment (Thomas-Gelf) [inline library/Jira/RestApi.php:130]: Fetching the setting and setting the default in multiple places feels redundant, you can address this if you want - but I do not see it as a requirement to get this merged.

**Review by dleusing — COMMENTED** [review]
Review finished

**Comment by fl0wx** [comment]
@Thomas-Gelf any updates on the review ?

**Comment by Thomas-Gelf** [comment]
@dleusing, @fl0wx: I'm sorry, I overlooked that this is now ready. Please re-check your config lookups, it seems that there are some inconsistencies, `field_icingaStatus` VS `icingaStatus` and `jira_key_fields` VS `ui`. The rest looks good to me. Once the lookups have been fixed, this should be ready for merge. It would also be great if you could squash (and force-push) your changes.

Thanks,
Thomas

**Review by dleusing — COMMENTED** [review]
reviewed

**Comment by dleusing** [comment]
changes are made

**Comment by fl0wx** [comment]
@Thomas-Gelf still anything open or can this be merged?

**Comment by fl0wx** [comment]
@Thomas-Gelf any updates?

**Comment by lippserd** [comment]
> @Thomas-Gelf any updates?

We'll have a look.

**Review by Thomas-Gelf — APPROVED** [review]
Now it looks good to me, thank you!

**Comment by puxilein** [comment]
@dleusing maybe as christmas present? :-)

**Comment by puxilein** [comment]
@dleusing any news?

**Comment by fl0wx** [comment]
@dleusing : when can we expect the requested rework?
@Thomas-Gelf @raviks789 : Would it be possible to merge it anyway if dleusing wont react?

**Comment by lippserd** [comment]
> @dleusing : when can we expect the requested rework? @Thomas-Gelf @raviks789 : Would it be possible to merge it anyway if dleusing wont react?

We can also make the changes.

**Comment by puxilein** [comment]
@lippserd that would be great! 👍

**Review by raviks789 — APPROVED** [review]

**Comment by fl0wx** [comment]
@Thomas-Gelf @raviks789 thanks a lot, can you also give an estimated time for the merge & release of a new version?

**Comment by puxilein** [comment]
Any news?

**Comment by puxilein** [comment]
*ping :-)
@Thomas-Gelf @raviks789 can you give an estimated time for the merge and release of the new version?

**Comment by fl0wx** [comment]
@lippserd @Thomas-Gelf  @raviks789 any update? 
"  thanks a lot, can you also give an estimated time for the merge & release of a new version? "

**Comment by lippserd** [comment]
Stop spamming.

**Pull request merged**